### PR TITLE
Refactor resource downloader

### PR DESCRIPTION
**Pull request by sclassen**
make UrlRequestResult unmodifiable and some code formatting

### CODE REVIEW

File: core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java [hunk 1]
```diff
@@ -461,7 +461,7 @@ private void markForDelete(URL location) {
             entry.unlock();
         }
     }
-    
+
     private void writeDownloadToFile(URL downloadLocation, InputStream in) throws IOException {
         byte buf[] = new byte[1024];
```
Comment (AndreasEhret) [inline core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java]: c-style declaration

File: core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java [hunk 2]
```diff
@@ -155,8 +156,8 @@ private void initializeFromURL(final UrlRequestResult location) throws IOExcepti
         CacheEntry entry = new CacheEntry(resource.getLocation(), resource.getRequestVersion());
         entry.lock();
         try {
-            resource.setDownloadLocation(location.URL);
-            final URLConnection connection = ConnectionFactory.getConnectionFactory().openConnection(location.URL); // this won't change so should be okay not-synchronized
+            resource.setDownloadLocation(location.redirectUrl);
+            final URLConnection connection = ConnectionFactory.getConnectionFactory().openConnection(location.redirectUrl); // this won't change so should be okay not-synchronized
             connection.addRequestProperty("Accept-Encoding", "pack200-gzip, gzip");
 
             File localFile = CacheUtil.getCacheFile(resource.getLocation(), resource.getDownloadVersion());
```
Comment (AndreasEhret) [inline core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java]: Line 164-165 condition is always false , maybe include this existing probable bug in the refactoring/cleanup
          Long size = location.length;
            if (size == null) {

File: core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java [hunk 3]
```diff
@@ -295,8 +296,8 @@ protected UrlRequestResult findBestUrl(final Resource resource) {
                     Map<String, String> requestProperties = new HashMap<>();
                     requestProperties.put("Accept-Encoding", "pack200-gzip, gzip");
```
Comment (hendrikebbers) [inline core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java]: we can directly create constants for the strings

Comment (sclassen) [reply core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java]: This is only step 1 of a larger refactoring.
Extractions of constants I will do later

File: core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java (hunk 2 above)
Comment (sclassen) [reply core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java]: Yes, have not touched that method yet

File: core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java (hunk 1 above)
Comment (sclassen) [reply core/src/main/java/net/sourceforge/jnlp/cache/ResourceDownloader.java]: done

**Pull request merged**